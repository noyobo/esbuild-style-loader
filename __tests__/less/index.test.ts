import path from 'path';
import { removeComments, runTest } from '../runTest';
import fse from 'fs-extra';

const OUTPUT_HTML = !!process.env.OUTPUT_HTML;

describe(path.basename(path.dirname(__filename)), function () {
  it('style-loader', async function () {
    const output = path.resolve(__dirname, 'output');
    fse.removeSync(output);
    const result = await runTest([path.resolve(__dirname, './index.jsx')], output);
    const allFile = result.outputFiles.map((item) => item.path);
    allFile.forEach((file) => {
      fse.ensureFileSync(file);
      let content = result.outputFiles.find((item) => item.path === file)?.text;
      content = removeComments(content);
      content = content.replace(/((style|css)-loader:\S+)/g, (match, loader, type) => {
        return `${type}-loader:${path.basename(loader)}`;
      });
      fse.writeFileSync(file, content);

      if (OUTPUT_HTML) {
        expect(1).toBe(1);
      } else {
        if (file.endsWith('.js') || file.endsWith('.css')) {
          expect(content).toMatchSnapshot();
        }
      }
    });
  });
});
