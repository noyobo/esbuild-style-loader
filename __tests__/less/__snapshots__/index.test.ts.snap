// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.test.ts style-loader: __tests__/less/output/index.css 1`] = `
"/* css-loader:./index.less */
.index__app {
  color: red;
  -webkit-user-select: none;
  user-select: none;
}
.index__title {
  font-size: 40px;
}
/*# sourceMappingURL=index.css.map */
"
`;

exports[`index.test.ts style-loader: __tests__/less/output/index.js 1`] = `
"(() => {
  var __create = Object.create;
  var __defProp = Object.defineProperty;
  var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
  var __getOwnPropNames = Object.getOwnPropertyNames;
  var __getProtoOf = Object.getPrototypeOf;
  var __hasOwnProp = Object.prototype.hasOwnProperty;
  var __require = /* @__PURE__ */ ((x) => typeof require !== "undefined" ? require : typeof Proxy !== "undefined" ? new Proxy(x, {
    get: (a, b) => (typeof require !== "undefined" ? require : a)[b]
  }) : x)(function(x) {
    if (typeof require !== "undefined")
      return require.apply(this, arguments);
    throw new Error('Dynamic require of "' + x + '" is not supported');
  });
  var __copyProps = (to, from, except, desc) => {
    if (from && typeof from === "object" || typeof from === "function") {
      for (let key of __getOwnPropNames(from))
        if (!__hasOwnProp.call(to, key) && key !== except)
          __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
    }
    return to;
  };
  var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
    // If the importer is in node compatibility mode or this is not an ESM
    // file that has been converted to a CommonJS file using a Babel-
    // compatible transform (i.e. "__esModule" has not been set), then set
    // "default" to the CommonJS "module.exports" for node compatibility.
    isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
    mod
  ));

  // style-loader:./index.less
  var index_default = { "app": "index__app", "title": "index__title" };

  // __tests__/less/index.jsx
  var import_react = __toESM(__require("react"));
  var import_client = __require("react-dom/client");
  var App = class extends import_react.default.Component {
    render() {
      return /* @__PURE__ */ import_react.default.createElement("div", { className: index_default.app }, /* @__PURE__ */ import_react.default.createElement("h1", { className: index_default.title }, "Hello World"));
    }
  };
  (0, import_client.createRoot)(document.getElementById("root")).render(/* @__PURE__ */ import_react.default.createElement(App, null));
})();
//# sourceMappingURL=index.js.map
"
`;
